src/comms/usb_hid.cpp

```diff
@@ -51,46 +51,56 @@ void HIDLayer::ping() {
 		// attempt to read
 		if (read()) {
 			// if we read, attempt to write
-
-			// set the packet to be written's ID
-			m_outgoingPacket.set_id((uint16_t)m_packetsSent);
 			if (!write())
 				Serial.printf("Failed to send ping %llu\n", m_packetsSent);
 		}
 	}
 }
 
-void HIDLayer::print() {
-	for (unsigned int i = 0; i < COMMS_PACKET_SIZE + 1; i++)
+void HIDLayer::print_outgoing() {
+	Serial.println("Outgoing packet:");
+	for (unsigned int i = 0; i < COMMS_PACKET_SIZE; i++)
 		Serial.printf("%.2x ", m_outgoingPacket.raw[i]);
 
 	Serial.println();
 }
 
+void HIDLayer::print_incoming() {
+	Serial.println("Incoming packet:");
+	for (unsigned int i = 0; i < COMMS_PACKET_SIZE; i++)
+		Serial.printf("%.2x ", m_incomingPacket.raw[i]);
+
+	Serial.println();
+}
+
 bool HIDLayer::read() {
 	// attempt to read a full packet
 	// this has no timeout
-	int bytes_read = usb_rawhid_recv(m_incommingPacket.raw, 0);
-	if (bytes_read == COMMS_PACKET_SIZE) {
+	int bytes_read = usb_rawhid_recv(m_incomingPacket.raw, 0);
+	if (bytes_read == COMMS_PACKET_SIZE)
+	{
 		// increment total number of packets read and return success
 		m_packetsRead++;
 		return true;
-	}
-	else {
+	} else {
 		return false;
 	}
 }
 
 bool HIDLayer::write() {
+	// verify that the first byte is set to something (0xff)
+	// prevents a weird comms issue where the whole packet is shifted left by one byte if the first byte is not ever set
+	m_outgoingPacket.raw[0] = 0xff;
+
 	// attempt to write a full packet
 	// this has a timeout, which is set to it's max value
-	int bytes_sent = usb_rawhid_send(m_outgoingPacket.raw, UINT16_MAX);
+	int bytes_sent = usb_rawhid_send(m_outgoingPacket.raw, 0);
 	if (bytes_sent == COMMS_PACKET_SIZE) {
 		// increment total number of packets sent and return success
 		m_packetsSent++;
 		return true;
-	} 
-	else {
+	} else {
+		Serial.println("Comms: failed write");
```

Comment:
will serial printing during execution cause problems (even if this is a failure log?)

Reply:
Nah thats fine.